pkg/nfs/configmap.go

```diff
@@ -0,0 +1,208 @@
+package nfs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"text/template"
+
+	storageosv1 "github.com/storageos/cluster-operator/pkg/apis/storageos/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// NFS server configuration constants.
+const (
+	DefaultAccessType = "readwrite"
+	DefaultSquash     = "none"
+	DefaultLogLevel   = "DEBUG"
+)
+
+func createConfig(instance *storageosv1.NFSServer) (string, error) {
+
+	// id needs to be unique for each export on the server node.
+	id := 57
+
+	var exports []string
+	// If no export list given, use defaults
+	if len(instance.Spec.Exports) == 0 {
+		exportCfg, err := exportConfig(id, instance.Name, DefaultAccessType, DefaultSquash)
+		if err != nil {
+			return "", err
+		}
+		exports = append(exports, exportCfg)
+	}
+
+	// Otherwise use export list
+	for _, export := range instance.Spec.Exports {
+		exportCfg, err := exportConfig(id, export.PersistentVolumeClaim.ClaimName, export.Server.AccessMode, export.Server.Squash)
+		if err != nil {
+			return "", err
+		}
+		exports = append(exports, exportCfg)
+		id++
+	}
```

Comment:
I think we can simplify this a bit as we've settled on having a single export per pv.  I don't think we will need to support multiple exports per server.

Reply:
Will remove this in a separate PR.

Reply:
Remove multiple exports support and changed it to customizable single export in this PR itself.